### PR TITLE
Add FAQ on causal messaging

### LINKED ISSUE #445: Add FAQ for "what is causal messaging"

### PR DESCRIPTION
**Pull request by SeanTAllen**
Closes #445

### CODE REVIEW

**Comment by SeanTAllen** [comment]
I'm not sure what a small example would look like. Do you have something in mind? If yes, please feel free to update in place or paste here and I can add.

**Comment by EpicEric** [comment]
I've tried to come up with an example but I figured it would only make the explanation more cumbersome. Re-reading the text, I believe the current explanation is good.

File: content/faq/index.md [hunk 1]
```diff
@@ -262,3 +262,11 @@ To add options to the link command, I would compile my program as something like
...
+
+When we say that Pony has causal **messaging**, we mean that the Pony runtime provides certain guarantees about message delivery order. Given two actors, actor A and actor B. All messages sent from actor A will arrive at actor B in the order they were sent by Actor A and will be processed by Actor B in the same order. The causal ordering between messages will be preserved as an invariant by the runtime.
+
+Causal messaging extends to a chain of actors. For example, if Actor B sends messages to Actor C in response to messages from Actor A, then there is a causal messaging chain from Actor A to Actor C. The cause of a message is a message from another actor and the ordering will be preserved.
+
+It is essential to understand that the causal ordering of messages only exists between two given actors. So for example, if actors X and Y send to actor Z, there is no guarantee what order the messages will be processed by Z except that we can say, all messages sent by X will be processed in the order they were sent, as will all messages sent by Y. However, messages from X and Y can arrive at Z interleaved in any order that maintains the causal ordering of X to Z and Y to Z.
```
Comment (jemc) [inline content/faq/index.md]: The statement "the causal ordering of messages only exists between two given actors" has the potential to appear contradictory to the earlier statement that "causal messaging extends to a chain of actors".

That is, I understand what you meant here and I understand that there is no contradiction, but your average reader may not. Here's a suggested revision that I think makes the meaning more clear:

```suggestion
It is essential to understand that the causal ordering of messages only exists between the two actors who are the sender and recipient of those messages. So for example, if actors X and Y send to actor Z, there is no guarantee what order the messages will be processed by Z except that we can say, all messages sent by X will be processed in the order they were sent, as will all messages sent by Y. However, messages from X and Y can arrive at Z interleaved in any order that maintains the causal ordering of X to Z and Y to Z.
```

**Comment by SeanTAllen** [comment]
@jemc suggestion applied.

@jemc @EpicEric let me know if you are good with the change. If you are, I'll squash and then merge.

**Pull request merged**